clustered-indexes.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Clustered Indexes
+summary: Learn how clustered indexes apply to TiDB.
+---
+
+# Clustered Indexes
+
+> **Warning:**
+>
+> Clustered indexes is an experimental feature introduced in TiDB 5.0-rc.
+
+Clustered indexes provide the ability to organize tables in a way that can improve the performance of certain queries. The term _clustered_ in this context refers to the _organization of how data is stored_ and not _a group of database servers working together_. Some database management systems may also refer to clustered indexes as _index-organized tables_ (IOT).
+
+TiDB only supports clustering by a table's `PRIMARY KEY`. With clustered indexes enabled, the terms _the_ `PRIMARY KEY` and _the clustered index_ might be used interchangeably.
```

Comment:
I wanted to say this because I think this is a concept that confuses new users. But I think the wording may still be a little clumsy (we use "might" in docs instead of "may" to avoid ambiguity, but it doesn't sound that natural to me.)

The `PRIMARY KEY` refers to the constraint versus clustered index refering to the physical implementation. It's not the same thing, but yes, in MySQL they are often used interchangeably.